tests/app/dao/test_uploads_dao.py

```diff
@@ -0,0 +1,168 @@
+
+
+def test_get_uploads_orders_by_created_at_desc(sample_template):
+    letter_template = create_precompiled_template(sample_template.service)
+
+    upload_1 = create_job(sample_template, processing_started=datetime.utcnow(),
+                          job_status=JOB_STATUS_IN_PROGRESS)
+    upload_2 = create_job(sample_template, processing_started=datetime.utcnow(),
+                          job_status=JOB_STATUS_IN_PROGRESS)
+    upload_3 = create_uploaded_letter(letter_template, sample_template.service, status='delivered')
+
+    results = dao_get_uploads_by_service_id(service_id=sample_template.service_id).items
+
+    assert len(results) == 3
+    assert results[0].id == upload_3.id
+    assert results[1].id == upload_2.id
+    assert results[2].id == upload_1.id
+
+
+def test_get_uploads_orders_by_processing_started_desc(sample_template):
+    days_ago = datetime.utcnow() - timedelta(days=3)
+    upload_1 = create_job(sample_template, processing_started=datetime.utcnow() - timedelta(days=1),
+                          created_at=days_ago,
+                          job_status=JOB_STATUS_IN_PROGRESS)
+    upload_2 = create_job(sample_template, processing_started=datetime.utcnow() - timedelta(days=2),
+                          created_at=days_ago,
+                          job_status=JOB_STATUS_IN_PROGRESS)
+
+    results = dao_get_uploads_by_service_id(service_id=sample_template.service_id).items
+
+    assert len(results) == 2
+    assert results[0].id == upload_1.id
+    assert results[1].id == upload_2.id
+
+
+def test_get_uploads_orders_by_processing_started_and_created_at_desc(sample_template):
+    letter_template = create_precompiled_template(sample_template.service)
+
+    days_ago = datetime.utcnow() - timedelta(days=4)
+    upload_0 = create_uploaded_letter(letter_template, service=letter_template.service)
+    upload_1 = create_job(sample_template, processing_started=datetime.utcnow() - timedelta(days=1),
+                          created_at=days_ago,
+                          job_status=JOB_STATUS_IN_PROGRESS)
+    upload_2 = create_job(sample_template, processing_started=datetime.utcnow() - timedelta(days=2),
+                          created_at=days_ago,
+                          job_status=JOB_STATUS_IN_PROGRESS)
+    upload_3 = create_uploaded_letter(letter_template, service=letter_template.service,
+                                      created_at=datetime.utcnow() - timedelta(days=3))
+
+    results = dao_get_uploads_by_service_id(service_id=sample_template.service_id).items
+
+    assert len(results) == 4
+    assert results[0].id == upload_0.id
+    assert results[1].id == upload_1.id
+    assert results[2].id == upload_2.id
+    assert results[3].id == upload_3.id
+
+
+def test_get_uploads_is_paginated(sample_template):
+    letter_template = create_precompiled_template(sample_template.service)
+
+    upload_1 = create_uploaded_letter(letter_template, sample_template.service, status='delivered',
```

Comment:
Optional: Your set up could be simpler here, we don't need many different types of uploads. Your test set up for `test_get_uploads_should_paginate` is much simpler and you could do similar here if you wanted